### ISSUE TITLE
Tailwind 4 compatibility

### ISSUE THREAD

**Issue by calebporzio** [livewire/flux#811]
This issue will house sub-issues related to Tailwind 4 compatibility

**Comment by nmfirdausw** [comment]
Had issue with searchable select here with Tailwind v4.

The list is not filtered when search, 
No results found is shown at the bottom,
And the check Icon show at above of the selection, not side by side

**Comment by joshhanley** [comment]
@nmfirdausw thanks for the info! Yep, see the linked issue #783 with what is going on. We will investigate it once it gets closer to the release incase it's just a bug with Tailwind 4.

**Comment by marksalmon** [comment]
@nmfirdausw If you want a quick and dirty fix while you're waiting, it can be done by publishing the select component and tweaking the following tailwind utility classes:

```diff
options.blade.php
- <ui-empty class="data-[hidden]:hidden">{{ $empty }}</ui-empty>
+ <ui-empty class="data-hidden:hidden!">{{ $empty }}</ui-empty>
``` 

```diff
option.blade.php

$classes = Flux::classes()
-   ->add('group/option overflow-hidden data-[hidden]:hidden group flex! items-center px-2 py-1.5 w-full focus:outline-none')
+   ->add('group/option overflow-hidden data-hidden:hidden! group flex! items-center px-2 py-1.5 w-full focus:outline-none')
```

```diff
empty.blade.php

$classes = Flux::classes()
-   ->add('data-[hidden]:hidden block items-center px-2 py-1.5 w-full')
+   ->add('data-hidden:hidden! block items-center px-2 py-1.5 w-full')

**Comment by nmfirdausw** [comment]
@marksalmon  Thanks. Will use that for a while.

**Comment by timkley** [comment]
FYI, Tailwind CSS v4 was now released: https://tailwindcss.com/blog/tailwindcss-v4

**Comment by joshhanley** [comment]
@timkley thanks!

**Comment by timgavin** [comment]
Just upgraded my app to Tailwind 4 and I am seeing a ton of issues with Flux. Not sure if it's on my end through. I removed the theme and went back to default, yet still seeing the issues. Modal is in the upper-left of the browser, switches are just solid colors, buttons blend into backgrounds, dark mode buttons are all smashed together to the left, etc.

Just wanted to put this here incase anyone else is having these freak outs. I'm hesitant to try and fix everything if it is indeed a Flux issue. If not, then I have some work to do!

**Comment by joshhanley** [comment]
@timgavin yeah Flux currnently isn't compatible with Tailwind V4. We've got a solution but it has breaking changes, so we're just working through that at the moment. See https://x.com/calebporzio/status/1881450082459070494

**Comment by timgavin** [comment]
> [@timgavin](https://github.com/timgavin) yeah Flux currnently isn't compatible with Tailwind V4. We've got a solution but it has breaking changes, so we're just working through that at the moment. See https://x.com/calebporzio/status/1881450082459070494

I looked on his account and totally missed that! thanks!

**Comment by jeffchown** [comment]
@cjmellor Per @joshhanley 's comment yesterday (https://github.com/livewire/flux/issues/811#issuecomment-2613725867), Flux is not yet Tailwind v4 compatible.

**Comment by joshhanley** [comment]
@cjmellor thanks! Yep, I found that dark mode issue when I was testing the other week, so it should be fixed once we release Tailwind v4 support.

**Comment by raz-iacob** [comment]
any idea when tailwind 4 support will be available?

**Comment by joshhanley** [comment]
@raz-iacob we're working on it actively at the moment, but it has breaking changes, so it's not a simple matter of just merging it. We will be providing some updates on this shortly.

**Comment by raz-iacob** [comment]
@joshhanley thank you!

**Comment by jeffchown** [comment]
UPDATE: Based on an email Caleb sent this morning, it looks like Tailwind 4 compatibility will be launched with Flux 2.0 on Feb. 19th along with new components!:

- Calendar
- Date picker
- Date range picker
- Line & Area Charts

**Comment by timgavin** [comment]
> UPDATE: Based on an email Caleb sent this morning, it looks like Tailwind 4 compatibility will be launched with Flux 2.0 on Feb. 19th along with new components!:
> 
> * Calendar
> * Date picker
> * Date range picker
> * Line & Area Charts

Good news! I was an idiot and upgraded Tailwind the day it was released and now everything's broken 😂😭

**Comment by PrinceMinky** [comment]
> UPDATE: Based on an email Caleb sent this morning, it looks like Tailwind 4 compatibility will be launched with Flux 2.0 on Feb. 19th along with new components!:
> 
> * Calendar
> * Date picker
> * Date range picker
> * Line & Area Charts

Just in time for Laravel 12! I'll do well to remember I have a family and not get lost in the excitement 💃
